hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -1431,9 +1431,19 @@ extends Importer with ucs.SplitElaborator:
      *  variables we found in `p`. */
     def arrow(lhs: Tree, rhs: Tree): Ctxl[Pattern] =
       val pattern = go(lhs)
-      val termCtx = ctx ++ pattern.variables.allocate
+      val variables = pattern.variables.allocate
+      // - `contextEntries` will be added to the context
```

Comment:
TODO: document why symbols are duplicated here; add to future minor improvements in tracking issue.

Reply:
Resolved in 28336d78a7cc23eb1504d7ffb96c115a0fa45961.